Makefile: add missing GRPC protos dependency

benesch opened the pull request:
Generating GRPC protos depends on protoc-gen-grpc-gateway, which is
installed by BOOTSTRAP_TARGET. That prerequisite wasn't properly
specified.

Release note: None

benesch committed:
Makefile: add missing GRPC protos dependency

Generating GRPC protos depends on protoc-gen-grpc-gateway, which is
installed by BOOTSTRAP_TARGET. That prerequisite wasn't properly
specified.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/21898)
<!-- Reviewable:end -->

benesch commented:
Fixes https://teamcity.cockroachdb.com/viewLog.html?buildId=497528&buildTypeId=Cockroach_UnitTests_BuildTestBinary&tab=buildLog.

benesch committed:
Makefile: compile protoc for host, not target

protoc needs to be compiled for the host, as it's invoked on the host to
produce generated protobufs during build time. We've been accidentally
cross-compiling it for the target. (This wasn't previously a problem
because `make protobuf` was not invoked automatically during builds, and
so was never triggered in a cross-compiling build.)

Release note: None

nvanbenschoten commented on Makefile:
```diff
@@ -447,7 +447,7 @@ CMAKE_SYSTEM_NAME := Windows
 
 CONFIGURE_FLAGS += --host=$(XHOST_TRIPLE) CC=$(CC_PATH) CXX=$(CXX_PATH)
-CMAKE_FLAGS += -DCMAKE_C_COMPILER=$(CC_PATH) -DCMAKE_CXX_COMPILER=$(CXX_PATH) -DCMAKE_SYSTEM_NAME=$(CMAKE_SYSTEM_NAME)
+XCMAKE_FLAGS := $(CMAKE_FLAGS) -DCMAKE_C_COMPILER=$(CC_PATH) -DCMAKE_CXX_COMPILER=$(CXX_PATH) -DCMAKE_SYSTEM_NAME=$(CMAKE_SYSTEM_NAME)
```
Is `XCMAKE_FLAGS` standard? If not, consider adding a comment explaining that the variable contains cross-compilation-specific flags.

benesch replied:
Not even a little bit standard, but then again none of this stuff is. Good point, though: a comment is always appreciated. Done!

mberhault commented:
LGTM. I'm guessing this is the reason for the nightly failures.

benesch commented:
Yerp. Thought I had hit the button last night, but no such luck.

On Tue, Jan 30, 2018 at 2:43 AM marc <notifications@github.com> wrote:

> LGTM. I'm guessing this is the reason for the nightly failures.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/21898#issuecomment-361503081>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA15IDvGZmV_lh62zErcuoMv8oQO3vuCks5tPsg6gaJpZM4Rxd-_>
> .
>
